.buildbot.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
+
+set -e
+
+# Buildbot will have done a shallow clone, but the formatting step requires
+# both full git history, and a complete set of upstream llvm release tags.
+git fetch --unshallow
+git fetch --tags https://github.com/llvm/llvm-project
+
+INST_DIR=`pwd`/inst
+
+mkdir -p build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=${INST_DIR} \
+    -DLLVM_INSTALL_UTILS=On \
+    -DCMAKE_BUILD_TYPE=release \
+    -DLLVM_ENABLE_ASSERTIONS=On \
+    -DLLVM_ENABLE_PROJECTS="lld;clang" \
+    ../llvm
+make -j `nproc` install
+
+# clang-format any new files that we've introduced ourselves.
+cd ..
+PATH=${INST_DIR}/bin:${PATH}
+sh yk_format_new_files.sh
```

Comment:
CI is still using the old formatter.

Reply:
Yeah, I realised this late last night.

Reply:
Should we use `clang/tools/clang-format/git-clang-format upstream/main` as well here?

Reply:
yes, but you will have to add `upstream` as a remote after the unshallow stage.

The `git fetch` can then use the upstream name instead of a raw url.

Reply:
We may not actually need an unshallow clone any more. May be worth trying without that also.